shuup_checkoutfi/models.py

```diff
@@ -80,16 +80,20 @@ def process_payment_return_request(self, service, order, request):
         if status in (2, 3, 5, 6, 8, 9, 10):
             if checkout.validate_payment_return(**fields):
                 payment_id = fields["payment"]
-                order.create_payment(
-                    order.taxful_total_price,
-                    payment_identifier="Checkout.fi %s" % payment_id,
-                    description=_(
-                        "Checkout.fi {payment_id} (ref {order_reference})"
-                    ).format(
-                        payment_id=payment_id,
-                        order_reference=fields["order_reference"]
+                try:
+                    order.create_payment(
+                        order.taxful_total_price,
+                        payment_identifier="Checkout.fi %s" % payment_id,
+                        description=_(
+                            "Checkout.fi {payment_id} (ref {order_reference})"
+                        ).format(
+                            payment_id=payment_id,
+                            order_reference=fields["order_reference"]
+                        )
                     )
-                )
+                # order already paid, move on..
```

Comment:
What if you actually check if the order is paid? If the order is not paid here we should likely raise here. Or at least log the error when the order is not fully paid. If you decide to use logging approach here you can check whether the `raven` is installed before capturing.

Reply:
I guess reason for this raising could very well be rounding issue too in some cases. But I guess it is safe to swallow the error when the order is actually fully paid.

Reply:
But non of those sentry errors indicates that this wouldn't be nothing more than just a duplicate call.

Reply:
I guess no need since core only raises if fully paid.